src/routes/v2/authenticatedSites/settings.js

```diff
@@ -77,11 +80,58 @@ class SettingsRouter {
     return next()
   }
 
+  async getRepoPassword(req, res, next) {
+    const { userWithSiteSessionData } = res.locals
+
+    const passwordRes = await this.settingsService.getEncryptedPassword(
+      userWithSiteSessionData
+    )
+
+    if (passwordRes.isErr()) {
+      throw passwordRes.error
```

Comment:
sanity check here: should we return some sort of 404 rather than thorwing an error here?

Reply:
Hmm this is thrown so that the error can be handled by our error handler instead - should we be catching individual errors here to return a specific error?

Reply:
Had a chat with Chin about this - given that it's difficult to step through to retrieve all the possible errors here now, I'm leaning towards leaving this as `throw` and fixing it after the mvp is out (by refactoring to ts) - do you have concerns about this?

Reply:
@alexanderleegs am ok with this, but are we sure that there is is no errors meant for internal consumption thrown back to the end user?

Reply:
Verified offline - the thrown isomer errors have user facing error messages, and any other error gets caught by our error handler to throw a generic error and message